scripts/json_parser.py

```diff
@@ -26,13 +27,20 @@
 """
 
 
-def fix_and_parse_json(    
+def extract_json_from_string(input_string: str) -> Dict[str, Any]:
+    json_pattern = regex.compile(r"\{(?:[^{}]|(?R))*\}")
+    json_string = json_pattern.search(input_string).group(0)
+    json_string = json_string.replace("\n", " ").replace("  ", " ")
```

Comment:
If we go down that rabbit hole, also include "\ n".
Google seems to be trolling with a space in between

Reply:
This will still error if no groups are found (AttributeError: 'NoneType' object has no attribute 'group')
Suggest you mod it to something like:

`
def extract_json_from_string(input_string: str) -> Dict[str, Any]:
    json_pattern = regex.compile(r"\{(?:[^{}]|(?R))*\}")
    json_string_groups = json_pattern.search(input_string)
    if len(json_string_groups) > 0:
        json_string = json_string_groups.group(0)
    else:
        json_string = ""
    json_string = json_string.replace("\n", " ").replace("  ", " ")
    return json_string
`

Reply:
or review the PR I created a few hours prior to that haha 
https://github.com/Torantulino/Auto-GPT/pull/697

Reply:
This PR can be easily deleted if there is a better solution :)